zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/SearchQueryResponseMapper.java

```diff
@@ -748,5 +775,34 @@ private static ProcessInstanceStateEnum toProtocolState(
     return ProcessInstanceStateEnum.fromValue(value.name());
   }
 
+  private static String determineValueType(final Object value) {
+    if (value instanceof Boolean) {
+      return TypeEnum.BOOLEAN.name();
+    }
+    if (value instanceof String) {
+      return TypeEnum.STRING.name();
+    }
+    if (value instanceof Integer || value instanceof Long) {
+      return TypeEnum.INT64.name();
+    }
+    if (value instanceof Float || value instanceof Double) {
+      return TypeEnum.FLOAT.name();
+    }
+
+    return TypeEnum.OBJECT.name(); // Fallback
+  }
+
+  private static String serializeValueToString(final Object value) {
```

Comment:
I have previously tried use GSON library over here, but was dealing with some problems from RDBMS, only RDBMS format for dates were serialising in invalid way, instead to use Strings.

So I kept it simple, and just use Strings, or escaped String once the object is instance of string.

Reply:
❓ Don't we have a JsonMapper we can inject into the REST controller and use here to serialize a value? This should be able to serialize a `string` value to an escaped `string`, I believe. I would have to take a closer look but I would guess we can reuse something here. On the other hand, do we need this? I am not entirely sure of the top of my head.

💭 Another way to look at it: if it's a `string` already, just leave it as is. Makes the serialization and deserialization a lot easier. Not sure if we're missing anything here but I think this would also just be fine.